pipeline_steps/common.groovy

```diff
@@ -1704,40 +1704,136 @@ void createComponentSkeleton(String name, String repoUrl, String jiraProjectKey)
   }
 }
 
-void createComponentGateTrigger(String name, String repoUrl, String jiraProjectKey){
-  repo_dir = "${WORKSPACE}/rpc-gating"
-  filename = "${name}.yml".replace("-", "_")
-  projectsFile = "${repo_dir}/rpc_jobs/${filename}"
-  withEnv(
-    [
-      "PROJECTS_FILE=${projectsFile}",
-      "REPO_DIR=${repo_dir}",
-      "COMPONENT_NAME=${name}",
-      "COMPONENT_REPO_URL=${repoUrl}",
-      "JIRA_PROJECT_KEY=${jiraProjectKey}",
-    ]
-  ){
-    withCredentials(
+void createComponentJobs(String name, String repoUrl, List releases, String jiraProjectKey){
+  String repo_dir = "${WORKSPACE}/rpc-gating-master"
+
+  dir(repo_dir) {
+    // NOTE(mattt): while likely to be unnecessary, we just re-clone rpc-gating
+    // here to ensure we're not committing anything sensitive that may have
+    // been created in ${WORKSPACE}/rpc-gating.
+    git branch: 'master', url: 'https://github.com/rcbops/rpc-gating'
+
+    createComponentGateTrigger(name, repoUrl, jiraProjectKey)
+    createComponentPreRelease(name, repoUrl, releases, jiraProjectKey)
+
+    withEnv(
       [
-        string(
-          credentialsId: 'rpc-jenkins-svc-github-pat',
-          variable: 'PAT'
-        ),
-        usernamePassword(
-          credentialsId: "jira_user_pass",
-          usernameVariable: "JIRA_USER",
-          passwordVariable: "JIRA_PASS"
-        ),
+        "ISSUE_SUMMARY=Add component skeleton to ${name}",
+        "ISSUE_DESCRIPTION=This issue was generated automatically as part of registering a new component.",
+        "LABELS=component-skeleton jenkins",
+        "JIRA_PROJECT_KEY=${jiraProjectKey}",
+        "TARGET_BRANCH=master",
+        "COMMIT_TITLE=Add default jobs to ${name}",
+        "COMMIT_MESSAGE=This project is being added to the RE platform. This change adds some\ndefault jobs required by all projects.",
       ]
     ){
-      sshagent (credentials:['rpc-jenkins-svc-github-ssh-key']){
-        sh """#!/bin/bash -xe
-          set +x; . ${WORKSPACE}/.venv/bin/activate; set -x
-          ${WORKSPACE}/rpc-gating/scripts/add_component_gate_trigger_job.sh
-        """
-      }
-    }
-  }
+      withCredentials(
+        [
+          string(
+            credentialsId: 'rpc-jenkins-svc-github-pat',
+            variable: 'PAT'
+          ),
+          usernamePassword(
+            credentialsId: "jira_user_pass",
+            usernameVariable: "JIRA_USER",
+            passwordVariable: "JIRA_PASS"
+          ),
+        ]
+      ){
+        sshagent (credentials:['rpc-jenkins-svc-github-ssh-key']){
+          sh """#!/bin/bash -xe
+            set +x; . ${WORKSPACE}/.venv/bin/activate; set -x
+            git status
+            git diff
+            #${WORKSPACE}/rpc-gating/scripts/commit_and_pull_request.sh
```

Comment:
We should probably have a common function that wraps the creds, ssh agent and shell call, as its used in a few places.